### PR TITLE
refactor all dependencies of AcousticDynamics to no longer use Grid

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Refactored dependencies of AcousticDynamics to use grid data container classes and GridIndexing instead of the Grid class.

Before completing refactors of AcousticDynamics, I'd like to containerize namelist variables, which is a fairly large task that should be separated into its own PR.

## Code changes:

- Refactored sim1_solver, pk3_halo, updatedzc, riem_solver3, and riem_solver_c to no longer use the Grid class
- Refactored internal methods in dyn_core.py such as _HaloUpdaters to no longer use Grid class

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: fv3core/stencils/dyn_core.py [hunk 1]
```diff
@@ -158,17 +164,6 @@ def get_nk_heat_dissipation(
     return nk_heat_dissipation
 
 
-def quantity_wrap(storage, dims: Sequence[str], grid_indexing: GridIndexing):
```
Comment (mcgibbon) [inline fv3core/stencils/dyn_core.py:161]: This has been moved to grid.py where it is needed by other code.

File: fv3core/stencils/dyn_core.py [hunk 2]
```diff
@@ -387,17 +389,23 @@ def __init__(
...
                 namelist.hord_tm,
                 self._dp_ref,
                 column_namelist,
                 d_sw.k_bounds(),
             )
-            self.riem_solver3 = RiemannSolver3(namelist)
-            self.riem_solver_c = RiemannSolverC(namelist)
+            self.riem_solver3 = RiemannSolver3(
+                grid_indexing,
+                spec.namelist.p_fac,
```
Comment (FlorianDeconinck) [inline fv3core/stencils/dyn_core.py:403]: A simple comment: as we expand our effort to rename and clarify the code, we might want to think about the relevance of the word `namelist` in a non-fortran code. I acknowledge that with time this has become a kind of climate simulation concept, but it's worth asking the question.
(One argument for keeping it being that it'll help not loose fortran dev)

**Comment by FlorianDeconinck** [comment]
LGTM - `fv_dynamics` needs to be updated

File: fv3core/stencils/dyn_core.py (hunk 2 above)
Comment (mcgibbon) [reply fv3core/stencils/dyn_core.py:403]: I agree, this will come up more as we refactor the namelist configuration.

**Review by FlorianDeconinck — APPROVED** [review]

**Pull request merged**